qubekit/torsions/fitting/forcebalance_wrapper.py

```diff
@@ -324,15 +325,28 @@ def _optimise(self, molecule: Ligand) -> Ligand:
             # back to fitting folder
             os.chdir(fitting_folder)
             # now we can make the optimize in file
-            self.generate_optimise_in(target_data=fitting_targets)
+            wq_port = self.generate_optimise_in(target_data=fitting_targets)
             # now make the forcefield file
             self.generate_forcefield(molecule=molecule)
 
             # now execute forcebalance
             with open("log.txt", "w") as log:
-                subprocess.run(
-                    "ForceBalance optimize.in", shell=True, stdout=log, stderr=log
+                import work_queue
+
+                workers = work_queue.Factory(
+                    "local", manager_host_port=f"localhost:{wq_port}"
                 )
+                # the optimisation takes space in gas, so 1 core for each OpenMM is good enough most likely
+                workers.cores = 1
+                # divide the memory for each worker
+                workers.memory = local_options.memory / local_options.cores
+                workers.min_workers = 1
+                workers.max_workers = local_options.cores
```

Comment:
Does this auto-scale the number of workers to match the total number of jobs? Should we make `workers.max_workers = local_options.cores - 1` so that one core is reserved for the ForceBalance server?

Reply:
it's tricky to say because the tests I have done show that the server almost always use very little cpu like 5%, and when it did use more the workers were done. 

Because the workers are auto-scaled, they are rarely actually using the quote of e.g. 10 cores, so maybe that's ok?

Reply:
but feel free to do  -1, I think you can modify the code in the PR, thanks